drivers/adc/ad7768-1/ad77681.c

```diff
@@ -494,16 +1620,25 @@ int32_t ad77681_soft_reset(struct ad77681_dev *dev)
  * @return 0 in case of success, negative error code otherwise.
  */
 int32_t ad77681_setup(struct ad77681_dev **device,
-		      struct ad77681_init_param init_param)
+		      struct ad77681_init_param init_param,
+		      struct ad77681_status_registers **status)
 {
 	struct ad77681_dev *dev;
-	int32_t ret = 0;
+	struct ad77681_status_registers *stat;
+	int32_t ret;
+	uint8_t scratchpad_check = 0xAD;
 
 	dev = (struct ad77681_dev *)malloc(sizeof(*dev));
 	if (!dev) {
 		return -1;
 	}
 
+	stat = (struct ad77681_status_registers *)malloc(sizeof(*stat));
+	if (!stat) {
+		free(stat);
```

Comment:
`free(dev)` instead of `free(stat)`.

This should fix the Codacy issue.

Reply:
When you reach this point in the code, only `dev` has memory successfully allocated. So before returning -1 it needs to be freed.

Reply:
Thanks. Done fixing.